src/components/registration-view/registration-view.jsx

```diff
@@ -0,0 +1,47 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+
+export function RegistrationView(props) {
+  const [ username, setUsername ] = useState('');
+  const [ password, setPassword ] = useState('');
+  const [ email, setEmail ] = useState('');
+  const [ birthday, setBirthday ] = useState('');
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+      console.log(username, password);
+      props.onRegistration(username);
+  };
+
+  return (
+    <form>
+      <label>
+        Username:
+        <input type="text" value={username} onChange={e => setUsername(e.target.value)} />
+      </label>
+      <label>
+        Password:
+        <input type="password" value={password} onChange={e => setPassword(e.target.value)} />
+      </label>
+      <label>
+        Email:
+        <input type="email" value={email} onChange={e => setEmail(e.target.value)} />
+      </label>
+      <label>
+        Birthday:
+        <input type="birthday" value={birthday} onChange={e => setBirthday(e.target.value)} />
+      </label>
+      <button type="submit" onClick={handleSubmit}>Register</button>
+    </form>
+  );
+}
+
+RegistrationView.propTypes = {
+  user: PropTypes.shape({
+    Username: PropTypes.string.isRequired,
+    Password: PropTypes.string.isRequired,
+    Email: PropTypes.string.isRequired,
+    Birthday: PropTypes.string.isRequired
+  }),
```

Comment:
I also dont think this component takes in a `user`. It has no use for it. It only takes in a function that is called when you click register. The user is what we pass out so shouldn't be among the props